test/functional/cmdline_options_tester/src/Test.java

```diff
@@ -409,7 +412,9 @@ public void dumpVerboseOutput(boolean result) {
 
 	public static int getUnixPID(Process process) throws Exception {
 		Class<?> cl = process.getClass();
-		if (!cl.getName().equals("java.lang.UNIXProcess")) {
+		if (!cl.getName().equals("java.lang.UNIXProcess") 
+		|| (!cl.getName().equals("java.lang.ProcessImpl") && isRiscv)  /* intended for Fedora_Linux/RISC-V */
```

Comment:
The check against ProcessImpl is not needed and is confusing, pls use `if (!cl.getName().equals("java.lang.UNIXProcess") || isRiscv)`

Reply:
Agree and fixed in this way.